src/Main.java

```diff
@@ -1,278 +1 @@
-package com.company;
 
-import javax.swing.text.StyledEditorKit;
-import java.util.Scanner;
-import java.io.File;
-import java.io.*;
-import java.util.Queue;
-import java.util.*;
-
-class Cell{
-    int x;
-    int y;
-    int gCost;
-    int hCost;
-    Cell parent;
-
-    public Cell(int x, int y, Cell parent){
-        this.x = x;
-        this.y = y;
-        this.parent = parent;
-    }
-
-    public Cell(int x, int y, Cell parent, int gCost, int hCost){
-        this.x = x;
-        this.y = y;
-        this.parent = parent;
-        this.gCost = gCost;
-        this.hCost = hCost;
-    }
-
-    public Cell getParent(){
-        return this.parent;
-    }
-}
-
-class MazeSolver {
-
-    //4 possible moves
-    static int[] move_x = {1, -1, 0, 0};
-    static int[] move_y = {0, 0, 1, -1};
-
-    // Using DFS algorithm, does not guarantee shortest path
```

Comment:
Thinking about it more, I think that your solution is incorrect if this is true. The point of the exercise is to find the shortest path no matter what. I think that in DFS, you *do* need to keep track of paths and discard the ones which are incorrect or longer until you have found the shortest path.